dist/samples/places-autocomplete-addressform/index.js

```diff
@@ -36,42 +48,56 @@ function initAutocomplete() {
 function fillInAddress() {
   // Get the place details from the autocomplete object.
   const place = autocomplete.getPlace();
-
-  for (const component in componentForm) {
-    document.getElementById(component).value = "";
-    document.getElementById(component).disabled = false;
-  }
+  let address1 = "";
+  let postcode = "";
 
   // Get each component of the address from the place details,
   // and then fill-in the corresponding field on the form.
   for (const component of place.address_components) {
     const addressType = component.types[0];
 
-    if (componentForm[addressType]) {
-      const val = component[componentForm[addressType]];
-      document.getElementById(addressType).value = val;
+    switch (addressType) {
+      case "street_number": {
+        address1 = component[componentLength[addressType]] + " " + address1;
+        break;
+      }
+
+      case "route": {
+        address1 += component[componentLength[addressType]];
+        break;
+      }
+
+      case "postal_code": {
+        postcode = component[componentLength[addressType]] + postcode;
+        break;
+      }
+
+      case "postal_code_suffix": {
+        postcode += "-" + component[componentLength[addressType]];
```

Comment:
this loop with `+=` concats seems to assume components has an order - but is there a guaranteed order of components?

Reply:
We cannot guarantee that the components from the response are provided in the "right" order so we have to handle both. Are you asking whether Places API guarantees that `postal_code` and `postal_code_suffix` components are properly populated for every address (i.e. no main codes entered as suffixes or vice versa)? I think we have to trust that they are.

- **Encounter `postal_code` before `postal_code_suffix`**
    1. `postcode` = `94043`
    2. `postcode` = `94043-1351` gets entered into the `postalField`.
- **Encounter `postal_code_suffix` before `postal_code`**
    1. `postcode` = `=1351`
    2. `postcode` = `94043-1351` gets entered into the `postalField`.
- **Encounter `postal_code` without `postal_code_suffix` present in response**
    1. `postcode` = `94043` gets entered into the `postalField`.
- **Encounter `postal_code_suffix` in response without `postal_code` present**
    1. `postcode` = `-1351` gets entered into the `postalField`. Transparent reflection of bad quality data.